Sources_v3/Database/DatabaseSession_Tests.swift

```diff
@@ -188,4 +188,58 @@ class DatabaseSession_Tests: StressTestCase {
             XCTAssertTrue(error is ClientError.CurrentUserDoesNotExist)
         }
     }
+    
+    func test_saveEvent_takesRemembersEventDate() throws {
+        // Create event payload with specific `createdAt` date
+        let eventPayload = EventPayload<DefaultDataTypes>(
+            eventType: .messageNew,
+            connectionId: .unique,
+            cid: .unique,
+            currentUser: .dummy(
+                userId: .unique,
+                role: .user,
+                unreadCount: nil
+            ),
+            unreadCount: .dummy,
+            createdAt: .unique
+        )
+        
+        // Save event to the database
+        try database.writeSynchronously { session in
+            try session.saveEvent(payload: eventPayload)
+        }
+        
+        // Load current user
+        let currentUser = database.viewContext.currentUser()
+        
+        // Assert `eventPayload.createdAt` is taked as last received event date
+        XCTAssertEqual(currentUser?.lastReceivedEventDate, eventPayload.createdAt)
+    }
+    
+    func test_saveEvent_fallbacksToCurrentDate_ifEventCreationDateIsUnknown() throws {
```

Comment:
... so this will be `test_saveEvent_doesntResetLastReceivedEventDate_whenEventCreatedAtValueIsNil`